### PR TITLE
Thingpeida integration test using k8s

### PR DESCRIPTION
**Pull request by jgd5**
This PR sets up a k8s cluster on Travis using [kind](https://kind.sigs.k8s.io/) . 
Most of the tests are ran inside the frontend pod.

### CODE REVIEW

File: tests/thingpedia-integration/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM localhost/almond-cloud
+
+USER root
+
+COPY --chown=almond-cloud:almond-cloud ./tests /opt/almond-cloud/tests
+
+RUN curl -sL https://rpmfind.net/linux/epel/7/x86_64/Packages/l/libpmem-1.1-1.el7.x86_64.rpm  -o libpmem-1.1-1.el7.x86_64.rpm && \
+    curl -sS https://downloads.mariadb.com/MariaDB/mariadb_repo_setup | bash && \
+    dnf -y install libpmem-1.1-1.el7.x86_64.rpm  MariaDB-client && \
+    rm -f libpmem-1.1-1.el7.x86_64.rpm && rm -rf /var/cache/dnf
```
Comment (gcampax) [inline tests/thingpedia-integration/Dockerfile:10]: Why do we need to install mariadb from third party repos? The base image OS (RHEL 8 UBI) should have mariadb in the normal repos.

File: tests/thingpedia-integration/run-integration.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+# Run thingpedia-integration test inside frontend pod
+POD=`kubectl get pod -l app=frontend -o jsonpath="{.items[0].metadata.name}"`
+kubectl exec $POD --  bash -c "cd /opt/almond-cloud && npx nyc tests/thingpedia-integration/thingpedia-integration.sh"
+
+# Run selenium test on ubuntu with Firefox installed
+THINGENGINE_CONFIGDIR=tests/thingpedia-integration/k8s npx nyc ts-node tests/test_website_selenium.js
+
+# Copy coverage outputs from frontend pod
+kubectl cp $POD:/opt/almond-cloud/.nyc_output .nyc_output
```
Comment (gcampax) [inline tests/thingpedia-integration/run-integration.sh:15]: Is /opt/almond-cloud writable from inside the test docker? Do we get any coverage at all from this process?

And what about Go coverage?

**Review by gcampax — APPROVED** [review]
Great job! This looks very nice, I'm looking forward to automatic testing of our k8s stuff before we deploy to staging. I have left a couple comments but overall I like this a lot.

File: tests/thingpedia-integration/k8s/config.d/config.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+DATABASE_URL: "mysql://root:password@mariadb.default.svc.cluster.local/thingengine_test?charset=utf8mb4_bin"
+DATABASE_PROXY_URL: "http://dbproxy.default.svc.cluster.local:8200"
+SERVER_ORIGIN: "http://localhost:8080"
```
[the remark this answers is not in the document]
Comment (jgd5) [reply tests/thingpedia-integration/k8s/config.d/config.yaml:3]: The localhost:8080 is port mapped to the frontend node port for the selenium tests.  Not sure how to get firefox to run inside REHL 8 UBI.  The other tests are ran locally inside the frontend pod so localhost:8080 also works.

File: tests/thingpedia-integration/Dockerfile (hunk 1 above)
Comment (jgd5) [reply tests/thingpedia-integration/Dockerfile:10]: Please let me know how to get it. I couldn't find it. I also did a dnf search mariadb .

File: tests/thingpedia-integration/run-integration.sh (hunk 2 above)
Comment (jgd5) [reply tests/thingpedia-integration/run-integration.sh:15]: It is in the frontend pod since I enabled root.  I think these coverage outputs are from the thingpedia  integration tests.
Go coverage is only enabled in the GO unittests.

File: tests/thingpedia-integration/k8s/config.d/stanford.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
+// -*- mode: js; indent-tabs-mode: nil; js-basic-offset: 4 -*-
```
[the remark this answers is not in the document]
Comment (jgd5) [reply tests/thingpedia-integration/k8s/config.d/stanford.js]: Kustomize doesn't like symlink outside of base dir.  I've merged stanford.js with config.yaml.

File: tests/thingpedia-integration/k8s/config.d/config.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (gcampax) [reply tests/thingpedia-integration/k8s/config.d/config.yaml:3]: No running firefox from outside makes sense, let Travis deal with configuring it...

File: tests/thingpedia-integration/Dockerfile (hunk 1 above)
Comment (gcampax) [reply tests/thingpedia-integration/Dockerfile:10]: Oh. I just looked too, mariadb is not part of the UBI images. This is fine then.

File: tests/thingpedia-integration/run-integration.sh (hunk 2 above)
Comment (gcampax) [reply tests/thingpedia-integration/run-integration.sh:15]: Yeah coverage of the test code itself is not very interesting. It would be nice to enable coverage of the actual frontend process. Not sure how we do that, and how we kill the frontend and capture it...

File: tests/website/test_my_api.js [hunk 5]
```diff
@@ -359,7 +359,7 @@ async function testMyApiConverse(auth) {
         }, {
             id: 4,
             type: 'text',
-            text: 'Sorry, I did not understand that. Can you rephrase it?',
+            text: 'Sorry, there seems to be a problem with the Genie service at the moment. Please try again later.',
```
Comment (gcampax) [inline tests/website/test_my_api.js:362]: Hold on from making this change. The NLP service is having some issue (https://github.com/stanford-oval/genienlp/issues/174) and is not staying up, but this is not the right reply.

File: tests/thingpedia-integration/run-integration.sh (hunk 2 above)
Comment (jgd5) [reply tests/thingpedia-integration/run-integration.sh]: Got the coverage outputs of frontend and worker.  Now the coverage report looks reasonable.
